hilti/toolchain/src/compiler/global-optimizer.cc

```diff
@@ -37,35 +42,86 @@ std::optional<std::pair<ModuleID, StructID>> typeID(T&& x) {
     return {{id->sub(-2), id->sub(-1)}};
 }
 
+// Helper function to determine whether a hook should be stored.
```

Comment:
Looking at where this is used, I can't quite tell what situation this is catching: in what situation's is it not "good"?

Reply:
You are right, this never returns `false` anymore. Removing it.